### PR TITLE
fix(prometheus): harden Prometheus against user errors

### PR DESCRIPTION
**Pull request by sjinks**
## Description

This PR makes the Prometheus plugin more tolerant of unexpected data.

It takes a different approach than #4772 to improve code readability.

### Plugin Updated: Prometheus

Make the Prometheus plugin more tolerant of unexpected data.

## Pre-review checklist

Please make sure the items below have been covered before requesting a review:

- [ ] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- [ ] This change uses a rollout method to ease with deployment (if applicable - especially for large scale actions that require writes).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

## Pre-deploy checklist

- [ ] VIP staff: Ensure any alerts added/updated conform to internal standards (see internal documentation). 

## Steps to Test

There are unit tests for this change; they should pass.

### CODE REVIEW

File: prometheus/inc/class-safeadapter.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	 *
+	 * @throws StorageException
+	 */
+	public function wipeStorage(): void {
+		$this->wrapped->wipeStorage();
+	}
+
+	private function invoke_method( string $method, array $data ): void {
+		if ( isset( $data['labelValues'] ) ) {
+			$data['labelValues'] = array_map( 'strval', $data['labelValues'] );
```
Comment (sjinks) [inline prometheus/inc/class-safeadapter.php:62]: Comment out this line to make the tests fail.

File: prometheus/inc/class-plugin.php [hunk 2]
```diff
@@ -114,11 +115,7 @@ public function process_metrics(): void {
 			return;
 		}
 
-		foreach ( $this->collectors as $collector ) {
-			if ( is_callable( [ $collector, 'process_metrics' ] ) ) {
-				$collector->process_metrics();
-			}
-		}
+		array_walk( $this->collectors, fn ( CollectorInterface $collector ) => $collector->process_metrics() );
```
Comment (WPprodigy) [inline prometheus/inc/class-plugin.php:118]: We're sure about removing the safeguard?

**Review by WPprodigy — APPROVED** [review]

File: prometheus/inc/class-plugin.php (hunk 2 above)
Comment (sjinks) [reply prometheus/inc/class-plugin.php:118]: Yes. `process_metrics()` is the part of the interface: https://github.com/Automattic/vip-go-mu-plugins/pull/4777/files#diff-570d0312e081d70141767fdbc5ab1cef3561f42de78e644998fdb4b1f77b0cf7R20

Comment (WPprodigy) [reply prometheus/inc/class-plugin.php:118]: Ah right, goes alongside the other change to add that back to the interface 👍

**Pull request merged**